### ISSUE TITLE
BUG: Black screen with 3.0.1 (Xorg) Video driver issue?

### ISSUE THREAD

**Issue by runningopenloop** [ezQuake/ezquake-source#179]
I'm running arch linux with an NVIDIA Quadro FX 3450 (NV42GL); nvidia proprietary drivers.  I'm able to run ezQuake 2.2 (ezquake-gl_linux-x64) I downloaded from the website but not 3.0.1 (ezquake-gl_linux-x64).  I installed 3.0.1 using nQuake.

Did something change between 2.2 and 3.0.1 which could impact video?  My conres and FOV is all messed up, but this seems to be associated with some changes to how the graphics are handled based what I read in some 2.2 to 3 upgrade notes.  Therefore also causing some video issue?

When running 3.0.1 I get a black screen with an the X windows "X" cursor.  I'm able to fire off a new Xterm (I'm running twm) then kill the quake process with a -9 flag.  Then I see repeated libpng warnings: iCCP: known incorrect sRGB profile.

**Comment by meag** [comment]
For the conres/fov changes, please see this [wiki article](https://github.com/ezQuake/ezquake-source/wiki/Common-Issues#fov-doesnt-look-the-same-as-ezquake-22).

Not sure about the black screen, will leave that to someone more qualified on linux.  Can you control menu/console, get sound etc?  Or is it just taking ages to start up?  ezQuake 3.0 uses SDL2, so there have been major changes to back-end audio/video/input.

**Comment by mmavova** [comment]
Sorry, I do not follow. How do you know that your FOV is 
messed up, if you get a blank screen?

The libpng warnings are harmless, afaik.

**Comment by runningopenloop** [comment]
meag: that was the article I was reading about fov.

mmavova: I was able to get quake running using the 2.2 opengl binary.

So here's the good news, I've got things working.  The bad news is I don't fully understand what's going on.  I found some strange things out about my window manager (TWM)?

Playing around (banging head against wall), I found I *could* the 3.0.1 binary running.  Seemed if I ran the 2.2 binary, then the 3.0.1 binary.  Progress, actually it was a lot more obfuscated that than, but anyway.. however I think the process of doing this started to modify the default config.cfg.
  At one point, I was able to set windowed mode, 640x480 (i.e. don't use desktop resolution), and I could get ezquake 3.0.1 running solid every time.

So I'm using TWM in X.  If I deleted the ezquake/configs/config.cfg (no longer in windowed mode)...
  - There was another factor going on too... after launching the 3.0.1 binary, I'd get the black screen/box/window.  I tried changing my X resolution (Ctrl+Alt+Plus) and the black box would become smaller the size of my desktop resolution (I could still pan the virtual desktop at 1280x1024, even though for example I could see the edge of a black box at 800x600).
  - Whenever I clicked this black box (or other background) I'd get a TWM menu and if I exited TWM, it seemed that I got some partial screen of quake.
  - I eventually set my twmrc to be that in this post: https://bbs.archlinux.org/viewtopic.php?id=105450 search on twmrc to find the reference file.  **Now when launching X using this twmrc and I start quake using 3.0.1 binary, no issues.**

So there is something about how TWM is handling the windowing that causes the video mode set to fail.  And when it fails and I get this black window/box, I don't get any sound/control either.

Before I tried the twm rc above, I exited TWM so there was no window manager/anything running, then I could always start ezquake then also... (but the keyboard wouldn't work in ezquake).

So bottom line, I wonder if something could be handled better in ezquake or SDL2 to ensure the video is initialized loaded correctly?  This wasn't an issue with the 2.2 opengl binary.  Or maybe TWM is doing something really stupid and I should use another window manager?
   - From the post I provided the URL for above... that's what I'm trying to do... multi-head quake.

I'm using arch linux 4.8.4-1-ARCH with drivers: nvidia-304xx 304.132-1.

**Comment by rakasunka** [comment]
Same issue here, ezquake-3 won't start in fullscreen mode in TWM. With my config i can start ezquake 3 with twm, but i can't run fullscreen. My special made cfg works at windowed fullscreen (moving the window make it windowed fullscreen), but there's still some drawback. I wouldn't use it anyway since mouse isn't handled correctly in windowed mode, i've reported this before, but i just gave up and i play ezquake 3 in it's own xorg server instead, which works flawless.

My tip is to start another xorg server with blackbox with 
xinit /usr/bin/blackbox -- :1 
(in a tty)
and start ezquake 3 in there.

Regards

**Comment by mmavova** [comment]
Do you have multiple monitors?
My friend found a crash related to that.

On 10/30/2016 01:24 PM, rakasunka wrote:
> Same issue here, ezquake-3 won't start in fullscreen mode
> in TWM. With my config i can start ezquake 3 with twm,
> but i can't run fullscreen. My special made cfg works at
> windowed fullscreen (moving the window make it windowed
> fullscreen), but there's still some drawback. I wouldn't
> use it anyway since mouse isn't handled correctly in
> windowed mode, i've reported this before, but i just gave
> up and i play ezquake 3 in it's own xorg server instead,
> which works flawless.
>
> My tip is to start another xorg server with blackbox
> with xinit /usr/bin/blackbox -- :1 (in a tty) and start
> ezquake 3 in there.
>
> Regards
>
> — You are receiving this because you commented. Reply to
> this email directly, view it on GitHub
> <https://github.com/ezQuake/ezquake-source/issues/179#issuecomment-257164765>,
>
>
or mute the thread

**Comment by runningopenloop** [comment]
Try using the twmrc in the thread I linked to.  You should find that TWM will then launch fullscreen.  So I'm beyond the issues of running quake.

However I'm currently struggling with getting the dual monitor thing going.  I'm not having a crash..  I got both instances quake running, but then the mouse thing isn't working out.  I need to work a bit more on this.

In my effort for multi-head, I've considered trying to run the windows version with wine in hopes that it can deal with mouse capture better.  Unfortunately I'm struggling there with TWM is iconifying quake when I start the second instance.

In fact, before I updated the TWM config rc to the one referred in the post above, I had this happen to me once, where on starting quake, it started iconified with TWM.  Then clicking on the icon everything worked.  However in the dual head case with two quake instances running in wine; it gets confused about keyboard/mouse focus when this happens.  I think maybe if it didn't get iconified, it would work?

So maybe there are some more TWM options worth looking into.  I was looking for other WM which might handle multi-pointer X better, but not sure... IceWM maybe, it seems to have some special focus rules, but still not working yet either.

I'll try the blackbox thing, but the concern for me is I think if I have two X servers running (to get dual head quake), one won't be 3d accelerated.

73,
Timothy

**Comment by runningopenloop** [comment]
Here something for me to try next time I have a go at this:
https://www.reddit.com/r/archlinux/comments/1cxhqj/multiseat_gaming_guide/

73,
Timothy

**Comment by jite** [comment]
Thanks for reporting, will take a look as soon as I get time. In the mean time please read the wiki about issue reporting for future reference.

**Comment by meag** [comment]
If I'm reading the conversation correctly, it was an issue with TWM, or SDL2 & TWM?

Closing - please re-open if I've misunderstood.
